### PR TITLE
Issue 304 automate security

### PR DESCRIPTION
**Pull request by tdonaworth**
## What changed

- Added Semgrep Scans to CI
- Cleaned up any prior existing Semgrep findings

## Issue

#304

### CODE REVIEW

File: backend/Dockerfile [hunk 1]
```diff
@@ -1,19 +1,20 @@
 FROM python:3.9-slim
 RUN apt-get update && apt-get clean && rm -rf /var/lib/apt/lists/*
-WORKDIR /opt/local/
+RUN pip install --no-cache-dir --upgrade pip==22.2.2 pipenv==2022.7.24 && useradd -ms /bin/bash app
+USER app
+WORKDIR /home/app
 
-ENV PYTHONPATH=/opt/local/
+ENV PYTHONPATH=/home/app
```
Comment (jonnalley) [inline backend/Dockerfile:7]: is this just a preference thing or was that a finding?

**Review by jonnalley — APPROVED** [review]
all the security! :-)

File: backend/Dockerfile (hunk 1 above)
Comment (tdonaworth) [reply backend/Dockerfile:7]: preference, and mostly to ensure we're running in user-space. The overall fix was to add the `app` user. So running out of that user's home just ensures correct permissions, etc.

Comment (jonnalley) [reply backend/Dockerfile:7]: Yes indeed.  Splendid, thank you

**Pull request merged**